src/backend/dx11/src/device.rs

```diff
@@ -1069,7 +1089,14 @@ impl hal::Device<Backend> for Device {
         I::Item: Borrow<(&'a Memory, R)>,
         R: RangeArg<u64>,
     {
-        // TODO: flush?
+
+        // go through every range we wrote to
+        for range in ranges.into_iter() {
+            let &(memory, ref range) = range.borrow();
+            let range = *range.start().unwrap_or(&0)..*range.end().unwrap_or(&memory.size);
```

Comment:
we need a helper `memory.resolve(range)` function for ranges, like we do in Metal.